rules/instance_network_interface_rules.yaml

```diff
@@ -0,0 +1,13 @@
+rules:
```

Comment:
This yaml should be heavily commented to describe exactly what the format is, what the expectation is when configured, and how the scanner works. The sample config should be the best documentation on how someone else can build a scanner for their org.

Since this scanner is focused on two things, that needs to be very clear in the doc. It was difficult for me to parse the objective just from reading the code, and other users will not read the code.

Since the main use is to 1) Ensure instances are only running on networks created in allowed projects (using XPN). and 2) Ensure instances with external IPs are only running on whitelisted networks, those two things need to be clearly defined.

Most orgs may not be using XPN like you are, so they need to know if/when this auditor is valuable to them.

It may make sense to expand this to have more general purpose usage if you aren't running XPN, but I'm open to adding that down the road.